crates/oxc_linter/src/rules/unicorn/consistent_date_clone.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{AstNode, context::LintContext, rule::Rule};
+use oxc_ast::AstKind;
+use oxc_ast::ast::{Argument, Expression};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+fn consistent_date_clone_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unnecessary `.getTime()` call")
+        .with_help("Prefer passing `Date` directly to the constructor when cloning")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct ConsistentDateClone;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// The Date constructor can clone a `Date` object directly when passed as an argument, making
+    /// timestamp conversion unnecessary.
+    ///
+    /// ### Why is this bad?
+    /// Doing the timestamp conversion using `.getTime()` is unnecessary.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// new Date(date.getTime());
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// new Date(date);
+    /// ```
+    ConsistentDateClone,
+    unicorn,
+    style,
+    fix
+);
+
+impl Rule for ConsistentDateClone {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::NewExpression(expr) = &node.kind() {
+            if !(expr.callee.is_specific_id("Date")
+                && expr.arguments.len() == 1
+                && expr.type_parameters.is_none())
+            {
+                return;
+            }
+            if let Argument::CallExpression(expr) = &expr.arguments[0] {
+                if let Expression::StaticMemberExpression(callee) = &expr.callee {
+                    if callee.property.name.as_str() == "getTime"
+                        && expr.arguments.len() == 0
+                        && !expr.optional
+                        && !callee.optional
+                    {
+                        ctx.diagnostic_with_fix(
+                            consistent_date_clone_diagnostic(expr.span),
+                            |fixer| {
+                                fixer.delete_range(Span::new(
+                                    callee.object.span().end,
+                                    expr.span.end,
+                                ))
+                            },
+                        );
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        "new Date(date)",
+        "date.getTime()",
+        "new Date(...date.getTime())",
+        "new Date(getTime())",
+        "new Date(date.getTime(), extraArgument)",
+        "new Date(date.not_getTime())",
+        "new Date(date?.getTime())",
+        "new NotDate(date.getTime())",
+        "new Date(date[getTime]())",
+        "new Date(date.getTime(extraArgument))",
+        "Date(date.getTime())",
+        "new Date(
```

Comment:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/93f9bc27a5cbcfef31f0991a97da6ccb558bc1f0/test/consistent-date-clone.js#L19-L40

Can you add the ToDo too :)?